docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
+
+
+### The overall Sequence Diagram:
```

Comment:
<img width="662" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/86831302/afcf0802-7d7b-4e62-8435-24e38c2a6991">
It seems this SD has included the entire program works. If it will be better you break each part down like storage, Ui, etc..